addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SiteIsolationScanRule.java

```diff
@@ -207,6 +207,11 @@ protected String getString(String param) {
         protected String getHeader() {
             return HEADER;
         }
+
+        @Override
+        protected AlertBuilder alert(String evidence) {
+            return super.alert(evidence).setAlertRef(PLUGIN_ID + "-1");
```

Comment:
This should be done in `SiteIsolationHeaderScanRule`.

Reply:
But each rule is different.

Reply:
Pass the `alertRef` through the constructor.